src/sqlfluff/core/parser/segments/base.py

```diff
@@ -564,39 +566,45 @@ def _position_segments(
 
                 if start_point and end_point and start_point != end_point:
                     # We should construct a wider position marker.
-                    repositioned_seg.pos_marker = PositionMarker.from_points(
+                    new_position = PositionMarker.from_points(
                         start_point,
                         end_point,
                     )
                 # If we have start point (or if they were equal above),
                 # just apply start point.
                 elif start_point:
-                    repositioned_seg.pos_marker = start_point
+                    new_position = start_point
                 # Do we have an end?
                 elif end_point:
-                    repositioned_seg.pos_marker = end_point
+                    new_position = end_point
                 else:  # pragma: no cover
                     raise ValueError("Unable to position new segment")
 
-            assert repositioned_seg.pos_marker  # hint for mypy
-            # Update the working position.
-            repositioned_seg.pos_marker = (
-                repositioned_seg.pos_marker.with_working_position(
-                    line_no,
-                    line_pos,
-                )
-            )
-            line_no, line_pos = repositioned_seg.pos_marker.infer_next_position(
-                repositioned_seg.raw, line_no, line_pos
+            assert new_position
+
+            # Otherwise update the working position.
+            new_position = new_position.with_working_position(line_no, line_pos)
```

Comment:
is this always going to run, since not under an `else` (matching `if` on line 543?) Was that the intention?

Reply:
Hmmm, good point. I don't think it needs to. I'll change that!

Reply:
I checked 😢 . Unfortunately this is still needed to be here. I tried moving it up and the tests fail. I'll leave it where it is for now.